frame/contracts/src/weights.rs

```diff
@@ -153,1251 +153,1321 @@ pub trait WeightInfo {
+			// Standard Error: 353_000
+			.saturating_add((2_325_912_000 as Weight).saturating_mul(c as Weight))
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	fn update_schedule() -> Weight {
-		(29_795_000 as Weight)
+		(28_396_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	fn instantiate_with_code(c: u32, s: u32, ) -> Weight {
-		(225_834_000 as Weight)
-			// Standard Error: 144_000
-			.saturating_add((165_632_000 as Weight).saturating_mul(c as Weight))
-			// Standard Error: 9_000
-			.saturating_add((2_563_000 as Weight).saturating_mul(s as Weight))
+		(130_743_000 as Weight)
+			// Standard Error: 318_000
+			.saturating_add((2_383_759_000 as Weight).saturating_mul(c as Weight))
+			// Standard Error: 20_000
+			.saturating_add((3_005_000 as Weight).saturating_mul(s as Weight))
 			.saturating_add(T::DbWeight::get().reads(6 as Weight))
-			.saturating_add(T::DbWeight::get().writes(5 as Weight))
+			.saturating_add(T::DbWeight::get().writes(4 as Weight))
 	}
 	fn instantiate(c: u32, s: u32, ) -> Weight {
-		(190_482_000 as Weight)
+		(207_803_000 as Weight)
 			// Standard Error: 12_000
-			.saturating_add((8_724_000 as Weight).saturating_mul(c as Weight))
+			.saturating_add((8_546_000 as Weight).saturating_mul(c as Weight))
 			// Standard Error: 1_000
-			.saturating_add((2_512_000 as Weight).saturating_mul(s as Weight))
+			.saturating_add((2_936_000 as Weight).saturating_mul(s as Weight))
 			.saturating_add(T::DbWeight::get().reads(6 as Weight))
-			.saturating_add(T::DbWeight::get().writes(4 as Weight))
+			.saturating_add(T::DbWeight::get().writes(3 as Weight))
 	}
 	fn call(c: u32, ) -> Weight {
-		(195_414_000 as Weight)
-			// Standard Error: 2_000
-			.saturating_add((3_920_000 as Weight).saturating_mul(c as Weight))
+		(195_899_000 as Weight)
+			// Standard Error: 5_000
+			.saturating_add((3_953_000 as Weight).saturating_mul(c as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 			.saturating_add(T::DbWeight::get().writes(2 as Weight))
 	}
 	fn claim_surcharge(c: u32, ) -> Weight {
-		(336_867_000 as Weight)
-			// Standard Error: 10_000
-			.saturating_add((5_262_000 as Weight).saturating_mul(c as Weight))
+		(130_447_000 as Weight)
+			// Standard Error: 2_000
+			.saturating_add((4_724_000 as Weight).saturating_mul(c as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 			.saturating_add(T::DbWeight::get().writes(4 as Weight))
 	}
 	fn seal_caller(r: u32, ) -> Weight {
-		(143_935_000 as Weight)
+		(140_274_000 as Weight)
 			// Standard Error: 128_000
-			.saturating_add((266_876_000 as Weight).saturating_mul(r as Weight))
+			.saturating_add((258_921_000 as Weight).saturating_mul(r as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
+			.saturating_add(T::DbWeight::get().writes(1 as Weight))
```

Comment:
Please note that these additional writes on contract callable functions is not per `r` and therefore disregarded. This is because the contract info data structure is now stored lazily and unconditionally once at the end of execution. Instead of eagerly and on demand (and potentially multiple times).